### PR TITLE
TW-1469: Fix the cursor automatically focus when changing reply a message

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket

- #1469

### Resolved

- SAFARI

https://github.com/linagora/twake-on-matrix/assets/99852347/394c5d13-6b91-41b9-8037-78e84f5a8ab4

- FIREFOX

https://github.com/linagora/twake-on-matrix/assets/99852347/2341924a-4e22-4b28-8c23-8a21bff0a40e

- CHROME

https://github.com/linagora/twake-on-matrix/assets/99852347/55ecb705-8cde-4216-bb3d-707c3d24ed13

### CODE REVIEW

**Review by Te-Z — APPROVED** [review]

File: lib/pages/chat/chat.dart [hunk 1]
```diff
@@ -1632,6 +1619,42 @@ class ChatController extends State<Chat>
     _currentChatScrollState = ChatScrollState.scrolling;
   }
 
+  void _requestInputFocus() {
+    if (!inputFocus.hasPrimaryFocus) {
+      inputFocus.requestFocus();
+    }
+  }
+
+  void _updateReplyEvent(Event? event) {
```
Comment (hoangdat) [inline lib/pages/chat/chat.dart:1628]: how aboout?
```suggestion
  void _updateReplyEvent({Event? event}) {
```

Comment (nqhhdev) [reply lib/pages/chat/chat.dart]: Good idea

File: lib/pages/chat/chat.dart [hunk 2]
```diff
@@ -1632,6 +1621,42 @@ class ChatController extends State<Chat>
...
+        'Chat::_updateReplyEvent():: FlutterError: $e',
+        e.stackTrace,
+      );
+    }
+  }
+
+  void _addSelectEvent(List<Event> events) {
+    setState(() {
+      selectedEvents.addAll(events);
+    });
```
Comment (sherlockvn) [inline lib/pages/chat/chat.dart:1644]: still need setState ?

Comment (sherlockvn) [reply lib/pages/chat/chat.dart:1644]: try ListNotifier

**Comment by sherlockvn** [comment]
still have problem in focus

https://github.com/linagora/twake-on-matrix/assets/43041967/d29e3f6f-58da-41dc-a875-1e050924c787

File: lib/pages/chat/chat.dart (hunk 2 above)
Comment (nqhhdev) [reply lib/pages/chat/chat.dart:1644]: In my PR don't focus on refactor `selectedEvents `. I only focus change reply event. Improve this later

**Comment by hoangdat** [comment]
please reabse, update ADR

**Review by hoangdat — APPROVED** [review]

**Pull request merged**